src/utils/utils.ts

```diff
@@ -325,6 +325,10 @@ export module _Util {
 
       return "#" + rHex + gHex + bHex;
     }
+
+    export function pointDistance(p1: Point, p2: Point) {
+      return Math.sqrt(Math.pow(p2.y - p1.y, 2) - Math.pow(p2.x - p1.x, 2));
```

Comment:
We could just use squared distances, saving some computation time. I also wonder if direct multiplcation is faster than `Math.pow()`; Can we test that?